### PR TITLE
Server-Tests

### PR DESCRIPTION
**Pull request by JRJurman**
This PR introduces server tests, the frameworks necessary for running server tests, and a little bit of package.json re-organizing to make the difference between dependencies and dev-dependencies more clear.

Anything required for running `npm start` (which requires building and starting the server and client), should be included in the dependencies. dev-dependencies will include libraries for doing development (watching for changes) and running tests (mocha and chai).

To run the tests, simply run `npm test`. To run only server or client side tests, run `npm run test:server` or `npm run test:client`

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -4,9 +4,14 @@
   "description": "",
   "main": "app.js",
   "dependencies": {
+    "babel": "^5.8.23",
+    "babelify": "^6.3.0",
```
Comment (chtinahow) [inline package.json]: Are we not doing the latest version?

Comment (JRJurman) [inline package.json]: We aren't, but the re-factoring to do so should be made in a separate PR (it's a decent amount of work). This change is just the move from dev-dependencies to dependencies.

**Comment by JRJurman** [comment]
Closing this PR in favor of a Rollup Centric Build PR, complete with Server AND Unit tests!

Coming soon to Repositories near you!
